src/pytroll_watchers/selector.py

```diff
@@ -0,0 +1,214 @@
+
+    Args:
+        selector_config: a dictionary of arguments to pass to the underlying redis instance, see
+          :py:class:`~redis.Redis`. You can also provide a ttl as an int
+          (seconds) or timedelta instance, otherwise it defaults to 300 seconds (5 minutes).
+        subscriber_config: a dictionary of arguments to pass to
+          :py:func:`~posttroll.subscriber.create_subscriber_from_dict_config`.
+
+    Yields:
+        JSON representations of posttroll messages.
+    """
+    subscriber = create_subscriber_from_dict_config(subscriber_config)
+
+    with closing(subscriber):
+        sel = TTLDict(**selector_config)
+        for msg in subscriber.recv():
+            key = msg.data["uid"]
+            try:
+                _ = sel[key]
+                logger.info(f"Discarded {str(msg)}")
+            except KeyError:
+                msg_string = str(msg)
+                sel[key] = msg_string
+                logger.info(f"New content {str(msg)}")
+                yield msg_string
+
+
+def _run_selector_with_managed_dict_server(selector_config, subscriber_config, publisher_config):
+    """Run the selector with a managed ttldict server."""
+    publisher = create_publisher_from_dict_config(publisher_config)
+    publisher.start()
+    with closing(publisher):
+        for msg in running_selector(selector_config, subscriber_config):
+            publisher.send(msg)
+
+
+def run_selector(selector_config, subscriber_config, publisher_config):
+    """Run the selector.
+
+    The aim of the selector is to skip messages that are duplicates to already published messages.
+    Duplicate in this context means messages referring to the same file (even if stored in different locations).
+
+    Messages that refer to new files will be published.
+
+    Args:
+        selector_config: a dictionary of arguments to pass to the underlying redis instance, see
+          :py:class:`~redis.Redis`. You can also provide a *ttl* for the
+          selector as an int (seconds) or timedelta instance, so that incoming messages are forgotten after that time.
+          If not provided, the ttl defaults to 300 seconds (5 minutes).
+          Also, you can provide a *directory* for the underlying datastructure to store the data in.
+        subscriber_config: a dictionary of arguments to pass to
+          :py:func:`~posttroll.subscriber.create_subscriber_from_dict_config`. The subscribtion is used as a source for
+          messages to process.
+        publisher_config: a dictionary of arguments to pass to
+          :py:func:`~posttroll.publisher.create_publisher_from_dict_config`. This publisher will send the selected
+          messages.
+
+    """
+    with _running_redis_server(port=selector_config.get("port"), directory=selector_config.pop("directory", None)):
+        _run_selector_with_managed_dict_server(selector_config, subscriber_config, publisher_config)
+
+
+@contextmanager
+def _running_redis_server(port=None, directory=None):
```

Comment:
```suggestion
def _start_redis_server(port=None, directory=None):
```

To me `running` sounds like we are just connecting to an already running instance, not starting a new server process.